File: 376_Wiggle_Subsequence.py
class Solution(object):
    def wiggleMaxLength(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        if not nums:
            return 0
        
        ans = 0
        last = 1
        count = 1
        flag = 0
        prev = nums[0]
            
        for n in nums:
            if prev > n and flag >= 0:
                flag = -1
                count += 1
            elif prev < n and flag <= 0:
                flag = 1
                count += 1
            ans = max(count, ans)
            prev = n
            
        return ans

File: 376_Wiggle_Subsequence.cpp
#include<algorithm>
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        
        if(nums.size()==0)
            return 0;
            
        int count = 1, flag = 0, prev = nums[0], ans = 0;
        for(int i=0; i<nums.size(); i++){
            if(prev>nums[i] && flag>=0){
                flag = -1;
                count++;
            } 
            else if(prev<nums[i] && flag<=0){
                flag = 1;
                count++;
            }
            ans = max(count, ans);
            prev = nums[i];
        }
        
        return ans;

    }
};
